### PR TITLE
New package:dunst

### PR DESCRIPTION
**Pull request by NoNameMan1231**
Still don't know if it works...

### CODE REVIEW

**Comment by NoNameMan1231** [comment]
Screenshot_20220422_193903

What's going on with my patch file

**Comment by Grimler91** [comment]
Use `./dunst-1.8.1/.valgrind.suppressions` or perhaps just `./.valgrind.suppressions` in patch header

File: x11-packages/dunst/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+TERMUX_PKG_HOMEPAGE=https://dunst-project.org/
+TERMUX_PKG_DESCRIPTION="Lightweight and customizable notification daemon"
+TERMUX_PKG_LICENSE="custom"
```
Comment (Grimler91) [inline x11-packages/dunst/build.sh:3]: License looks like BSD 3-Clause

**Comment by NoNameMan1231** [comment]
> Use `./dunst-1.8.1/.valgrind.suppressions` or perhaps just `./.valgrind.suppressions` in patch header

i will try it ... but now now. currently busy

**Comment by NoNameMan1231** [comment]
Looks like dunstrc.default can't patch. Should I don't patch it?

**Comment by NoNameMan1231** [comment]
```
ld: error: unable to find library -lrt
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:65: dunst] Error 1
make: *** Waiting for unfinished jobs....
Error: Process completed with exit code 2.
```

Damn , new type of error comes out

**Comment by thunder-coding** [comment]
> ```
> ld: error: unable to find library -lrt
> clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
> make: *** [Makefile:65: dunst] Error 1
> make: *** Waiting for unfinished jobs....
> Error: Process completed with exit code 2.
> ```
> 
> Damn , new type of error comes out 

You just need to strip the `-lrt` from [config.mk:40]https://github.com/dunst-project/dunst/blob/1d200c510b9e303cc74674f11fcde34b503f1feb/config.mk#L40()

**Comment by NoNameMan1231** [comment]
Done!

**Comment by NoNameMan1231** [comment]
Screenshot_20220424_192459
:(

**Comment by wmcbtech30** [comment]
`dbus-uuidgen > $PREFIX/lib/dbus/machine-id`

**Comment by NoNameMan1231** [comment]
bash: /data/data/com.termux/files/usr/lib/dbus/machine-id: No such file or directory

**Comment by wmcbtech30** [comment]
create a directory `/data/data/com.termux/files/usr/lib/dbus/` then rerun the command again. redirection does not work if you don't create a parent directory

**Comment by NoNameMan1231** [comment]
```
Unable to send notification: Cannot spawn a message bus without a machine-id: Unable to load /data/data/com.termux/files/usr/var/data/data/com.termux/files/usr/lib/dbus/machine-id or /data/data/com.termux/files/usr/etc/machine-id: Failed to open file ??/data/data/com.termux/files/usr/var/data/data/com.termux/files/usr/lib/dbus/machine-id??: No such file or directory

```

`/data/data/com.termux/files/usr/var/data/data/com.termux/files/usr/lib/dbus/machine-id` it's kinda weird...

**Comment by NoNameMan1231** [comment]
> ~~Create a directory `/data/data/com.termux/files/usr/etc/dbus/` then rerun the command again. redirection does not work if the parent directory does not exist~~
> 
> EDIT: The screenshot shows it tries to open it on nonexistent nested prefix directory which something is not right with your patch. it does search something at the correct path which i'm currently confused at.
> `dbus-uuidgen > $PREFIX/etc/machine-id`

It work!
But where did i messed up? Can you help me?

**Comment by wmcbtech30** [comment]
You may as well only patch the hardcoded paths which it doesn't respect https://github.com/dunst-project/dunst/blob/1d200c510b9e303cc74674f11fcde34b503f1feb/config.mk#L2 file. I think just settings the `config.mk` is sufficient enough

**Comment by NoNameMan1231** [comment]
> You may as well only patch the hardcoded paths if it doesn't respect https://github.com/dunst-project/dunst/blob/1d200c510b9e303cc74674f11fcde34b503f1feb/config.mk#L2 file. I think just setting the `config.mk` is sufficient enough



> You may as well only patch the hardcoded paths if it doesn't respect https://github.com/dunst-project/dunst/blob/1d200c510b9e303cc74674f11fcde34b503f1feb/config.mk#L2 file. I think just setting the `config.mk` is sufficient enough. and separate your patch files based on the files edited

So I need to rewrite patch again3?

**Comment by wmcbtech30** [comment]
Yes, but only modifying the `config.mk` file.

**Comment by NoNameMan1231** [comment]
> Yes, but only modifying the `config.mk` file.

Just config.mk? Don't patch other one?

**Comment by wmcbtech30** [comment]
Yes, but if it's not sufficient and errors rise when testing your artifacts. you may as well create an new individual patch file

**Comment by NoNameMan1231** [comment]
> 

Screenshot_20220424_194922
This only?

**Comment by NoNameMan1231** [comment]
> 

Screenshot_20220424_194922
This only?

**Comment by NoNameMan1231** [comment]
> Yes, but if it's not sufficient and errors rise when testing your artifacts. you may as well create an new individual patch file

Screenshot_20220424_195108
This only?

**Comment by NoNameMan1231** [comment]
> Yes, but if it's not sufficient and errors rise when testing your artifacts. you may as well create an new individual patch file



This only?

**Comment by wmcbtech30** [comment]
Yes but keep your patch filename informational. test your dunst package later on if its succeeds and reproduce the issue if possible

**Comment by NoNameMan1231** [comment]
Well, it work!

**Pull request merged**